### PR TITLE
[Grupa 1] Projekt 1 Milestone 2 - Kacper Grzymkowski Jan Gaska

### PR DESCRIPTION
**Pull request by Ja-Gk-00**

### CODE REVIEW

File: Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb [hunk 1]
```diff
@@ -0,0 +1,2102 @@
+{
```
Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Ciekawe byłoby sprawdzenie jak zmiana sposobu enkodowania wpływa na łatwość modelowania (one-hot oraz target encoding). Jest to analiza w celach dydaktycznych</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>To prawda - takie kodowania jest w tym wypadku zasadne - pytanie czy to uprościło zadanie. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Z uwagi na to, że zbiór jest niezbalansowany warto byłoby się zastanowić nad użyciem parametru stratify w ramach train_test_split</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Dobre, że opakowano kod w funkcje. <u>Skoro zbiór nie jest zbalansowany to być może lepiej niż analizować accuracy, byłoby stosować F1-score.</u></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Bardzo dobry tok myślenia. Super, że zastosowano precision-recall curve.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Bardzo dobra obserwacja :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Zabrakło jasnego komentarza, dlaczego jak zrobiliście</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Dobry pomysł - warto stawiać sobie takie pytania aby lepiej zrozumieć model i dane (Interpretowalne Uczenie Maszynowe)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>szkoda, że nie skorzystano z funkcji wcześniej zdefiniowanej</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (ptomaszewska) [inline Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb:1]: <p>Bardzo dobrze wykonany KM2 - widać umiejętności analityczne i chęć rozłożenia wszystkiego na czynniki pierwsze.</p><p> </p><p>W ramach KM3 warto byłoby wybrać optymalne parametry modeli (na zajęciach zostanie omówione strojenie). Należy uwzględnić kroswalidację.</p><p>Inne możliwe kierunki prac - sugestie:</p><ul><li>analiza drzew decyzyjnych pod kątem ich przycinania (tree pruning) - na zbiorze danych o politykach można to było zrobić na zasadzie prób i błędów  - można jednak do tego zagadnienia podejść w sposób bardziej analityczny</li><li>z uwagi na to, że zbiór jest stosunkowo łatwy a więc tworzone modele biorą pod uwagę niewielką  ilość zmiennych można się zastanowić nad wizualizacją granic decyzyjnych np. w SVM czy kNN</li><li>z uwagi na to, że zbiór jest niezbalansowany można pomyśleć nad zastosowaniem technik upsampling. Innym pomysłem jest zmiana progu odcięcia klas z domyślnej wartości 0.5 na inną</li><li>czy algorytmy do automatyczmej selekcji zmiennych wytypowałyby jako ważne te same zmienne co Wy zidentyfikowaliście?</li></ul><p> </p><p>Uwaga na przyszłość:</p><ul><li>Ciekawa byłaby analiza co jest "wyjątkowego" w obserwacjach, na których finalny model się myli (akurat w przypadku tego zbioru danych w dużym stopniu można to przewidzieć na podstawie EDA). Zajmuje się tym dziedzina Interpretowalnego Uczenia Maszynowego (być może będzie kiedyś okazja wspomnieć o wykorzystywanych algorytmach na zajęciach)</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>
